compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -331,6 +331,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    pub fn suggest_remove_last_method_call(
+        &self,
+        err: &mut Diagnostic,
+        expr: &hir::Expr<'tcx>,
+        expected: Ty<'tcx>,
+    ) -> bool {
+        if let hir::ExprKind::MethodCall(hir::PathSegment { ident: method, .. }, recv_expr, &[], _) = expr.kind &&
+            let Some(recv_ty) = self.typeck_results.borrow().expr_ty_opt(recv_expr) &&
+            self.can_coerce(recv_ty, expected) {
+                err.span_suggestion_verbose(
+                    expr.span.with_lo(method.span.lo() - rustc_span::BytePos(1)),
```

Comment:
Can you use `rcvr_expr.span.hi()` instead of subtracting 1 from `lo`? 

We can have spaces I think like `a . b()`

Reply:
Yes, I tried to use `rcvr_expr.span.hi()`,
but seems it's not right in the scenario of macro, for example:

```rust
let s: &str = format!("hello world").to_string()
```
the `rcvr_expr.span.hi()` is span after macro expansion.

Reply:
can you use Span::find_ancestor_inside?

Reply:
code updated, thanks!